experiments/multichain/multichain_integrated_client.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+from os import path, environ
+from sys import path as pythonpath
+import base64
+
+from gumby.experiments.dispersyclient import main
+from experiments.multichain.multichain_client import MultiChainDelayCommunity, MultiChainNoResponseCommunity
+from experiments.dispersy.hiddenservices_client import HiddenServicesClient
+
+from twisted.python.log import msg
+
+pythonpath.append(path.abspath(path.join(path.dirname(__file__), '..', '..', '..', "./tribler")))
+
+from Tribler.dispersy.candidate import Candidate
+from Tribler.community.multichain.community import MultiChainCommunity, MultiChainCommunityCrawler, MultiChainScheduler
+
+
+class MultiChainIntegratedClient(HiddenServicesClient):
+    """
+    Gumby client to start the MultiChain Community in conjunction with the HiddenServicesClient.
+    """
+
+    def __init__(self, *args, **kwargs):
+        HiddenServicesClient.__init__(self, *args, **kwargs)
+        msg("Starting MultiChain client")
+        """ Set the default MultiChainCommunity as community """
+        self._multichain_type = MultiChainCommunity
+        """ MultiChain is initialized later."""
+        self._multichain = None
+        self.vars['public_key'] = base64.encodestring(self.my_member_key)
+        """ Override the test files to speed up the test."""
+        # 10 Mb
+        self.test_file_size = 100 * 1024 * 1024
+        self.min_circuits = 1
+        self.max_circuits = 1
+
+    def registerCallbacks(self):
+        HiddenServicesClient.registerCallbacks(self)
+        self.scenario_runner.register(self.set_multichain_type, 'set_multichain_type')
+        self.scenario_runner.register(self.introduce_candidates, 'introduce_candidates')
+        self.scenario_runner.register(self.request_signature, 'request_signature')
+        self.scenario_runner.register(self.request_crawl, 'request_crawl')
+        self.scenario_runner.register(self.close, 'close')
+        """ Integrated callbacks"""
+        self.scenario_runner.register(self.start_multichain, 'start_multichain')
+        self.scenario_runner.register(self.start_scheduler, 'start_scheduler')
+
+    def set_multichain_type(self, multichain_type='MultiChainCommunity'):
+        """
+        Sets the community class of this gumby node to a special community.
+        """
+        msg("CommunityType: %s" % multichain_type)
+        if multichain_type == 'MultiChainDelayCommunity':
```

Comment:
What about having a common superclass, this class has duplicated code with MultichainClient.